src/middlewared/middlewared/plugins/network.py

```diff
@@ -1732,23 +1732,41 @@ async def sync(self, wait_dhcp=False):
             await self.middleware.call('interface.vlan_setup', vlan, disable_capabilities, parent_interfaces)
 
         bridges = await self.middleware.call('datastore.query', 'network.bridge')
-        for bridge in bridges:
-            name = bridge['interface']['int_interface']
+        # Considering a scenario where we have the network configuration
+        # physical iface -> vlan -> bridge
+        # If all these interfaces are set to have 9000 MTU, we won't be able to set that up on boot
+        # unless physical iface has a 9000 MTU when the bridge is being setup.
+        # To address such scenarios, we divide the approach in 4 steps:
+        # 1) Remove orphaned members from bridge
+        # 2) Sync all non-bridge interfaces ( in this order physical ifaces -> laggs -> vlans )
+        # 3) Setup bridge with MTU
+        # 4) Sync bridge interfaces
 
-            cloned_interfaces.append(name)
-            await self.middleware.call('interface.bridge_setup', bridge, sync_interface_opts)
+        for bridge in bridges:
+            await self.middleware.call('interface.pre_bridge_setup', bridge, sync_interface_opts)
 
-        self.logger.info('Interfaces in database: {}'.format(', '.join(interfaces) or 'NONE'))
-        # Configure VLAN before BRIDGE so MTU is configured in correct order
+        # Configure LAGG before VLAN so MTU is configured in correct order
```

Comment:
This is not only related to LAGG. Let's say `Configure VLAN interfaces MTU last as they are restricted by underlying interfaces MTU`